src/Xrefcheck/Config.hs

```diff
@@ -120,6 +123,7 @@ overrideConfig config
         , ncIgnoreAuthFailures        = overrideField ncIgnoreAuthFailures
         , ncDefaultRetryAfter         = overrideField ncDefaultRetryAfter
         , ncMaxRetries                = overrideField ncMaxRetries
```

Comment:
Should we rename this somehow? It can break old configs, is it important?

Reply:
Maybe `ncMaxTooManyRequestsRetries`, like this.
I think that in presence of `ncMaxTimeoutRetries` the name `ncMaxRetries` is confusing because from the lexical point of view its scope includes `ncMaxTimeoutRetries`, but in the code it's used only for 429 errors. So I think we should rename it.

Reply:
That makes sense. Let me think about this for a while more...

Reply:
After your comment about retrying on timeouts only for domains that earlier returned 429, I decided that it's better to use `maxRetries` field for all the errors, and `maxTimeoutRetries` only for timeouts. That is, we stop retrying if either total errors get greater than `maxRetries` or the total timeouts errors get greater than `maxTimeoutRetries`. Does it make sense?

Reply:
Sounds very sensible to me :+1: